datumaro/plugins/data_formats/datumaro_binary/mapper.py

```diff
@@ -0,0 +1,110 @@
+# Copyright (C) 2023 Intel Corporation
+#
+# SPDX-License-Identifier: MIT
+
+import struct
+from abc import ABC, abstractmethod
+from typing import Any, Dict, Tuple
+
+from datumaro.components.dataset_base import DatasetItem
+from datumaro.components.errors import DatumaroError
+from datumaro.components.media import Image, MediaElement, MediaType
+from datumaro.util import dump_json, parse_json
+
+
+class Mapper(ABC):
+    @staticmethod
+    @abstractmethod
+    def forward(obj: Any) -> bytes:
+        """Convert an object to bytes."""
+
+    @staticmethod
+    @abstractmethod
+    def backward(_bytes: bytes, offset: int = 0) -> Tuple[Any, int]:
+        """Build an object from bytes."""
+
+
+class StringMapper(Mapper):
+    @staticmethod
+    def forward(obj: str) -> bytes:
+        obj = obj.encode()
+        length = len(obj)
+        return struct.pack(f"I{length}s", length, obj)
+
+    @staticmethod
+    def backward(_bytes: bytes, offset: int = 0) -> Tuple[str, int]:
+        length = struct.unpack_from("I", _bytes, offset)[0]
+        offset += 4
+        string = struct.unpack_from(f"{length}s", _bytes, offset)[0].decode()
+        return string, offset + length
+
+
+class DictMapper(Mapper):
+    @staticmethod
+    def forward(obj: Dict[str, Any]) -> bytes:
+        msg = dump_json(obj)
+        length = len(msg)
+        return struct.pack(f"I{length}s", length, msg)
+
+    @staticmethod
+    def backward(_bytes: bytes, offset: int = 0) -> Tuple[Dict[str, Any], int]:
+        length = struct.unpack_from("I", _bytes, offset)[0]
+        offset += 4
+        parsed_dict = parse_json(_bytes[offset : offset + length])
+        return parsed_dict, offset + length
+
+
+class MediaMapper(Mapper):
+    ALLOWED_TYPES = {MediaType.IMAGE, MediaType.POINT_CLOUD}
+
+    @staticmethod
+    def forward(obj: MediaElement) -> bytes:
+        if obj.MEDIA_TYPE == MediaType.IMAGE:
+            size = obj.size if obj.has_size else (-1, -1)
+        elif obj.MEDIA_TYPE == MediaType.POINT_CLOUD:
+            size = (-1, -1)
```

Comment:
Is there any way to represent the size of point_cloud? it seems like (len(points), 3).
How about other media types?

Reply:
If it exists, please add NonImplementedError and TODO comments.

Reply:
Actually, it doesn't have `size` but have `path` and `extra_images`.
https://github.com/openvinotoolkit/datumaro/blob/12ff0d3bb159dec7746d48d44784d1bdf63c818d/datumaro/components/media.py#L527-L533

Therefore #830 will change this as follows.
https://github.com/openvinotoolkit/datumaro/blob/12ff0d3bb159dec7746d48d44784d1bdf63c818d/datumaro/plugins/data_formats/datumaro_binary/mapper/media.py#L88-L113